modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -171,6 +171,16 @@ const EditorUpload = function (editor: Editor): EditorUpload {
           }
         }
 
+        if (filteredResult.length > 0 || imagesToRemove.length > 0) {
```

Comment:
This doesn't seem right, as we wouldn't want to fire a change for `imagesToRemove` as that would be handled by the above `editor.undoManager.transact` call.

Also I'm pretty sure I told you the wrong thing here sorry... this probably needs to be before images are going to be removed. Otherwise we never get a change for the state between the images being updated and then the invalid ones being removed.